inspirehep/modules/hal/core/tei.py

```diff
@@ -0,0 +1,135 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2014-2017 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""HAL TEI Core."""
+
+from __future__ import absolute_import, division, print_function
+
+import langdetect
+from flask import render_template
+
+from inspirehep.utils.record import get_title
+
+from ..utils import (
+    get_abstract,
+    get_authors,
+    get_conference_city,
+    get_conference_country,
+    get_conference_date,
+    get_conference_record,
+    get_conference_title,
+    get_divulgation,
+    get_document_types,
+    get_doi,
+    get_domain,
+    get_journal_title,
+    get_language,
+    get_peer_reviewed,
+    get_publication_date,
+    is_published,
```

Comment:
Ideally all these should be moved into `inspire-schema` package, so that they can be naturally adjusted WRT schema changes.